// @smithy/signature-v4/dist-es/suite.fixture.js
export const region = "us-east-1";
export const service = "service";
export const credentials = {
    accessKeyId: "AKIDEXAMPLE",
    secretAccessKey: "REDACTED_AWS_SECRET",
};
export const signingDate = new Date("2015-08-30T12:36:00Z");
export const requests = [
    {
        name: "get-header-key-duplicate",
        request: {
            protocol: "https:",
            method: "GET",
            hostname: "example.amazonaws.com",
            query: {},
            headers: {
                host: "example.amazonaws.com",
                "my-header1": "value2,value2,value1",
                "x-amz-date": "20150830T123600Z",
            },
            path: "/",
        },
        authorization: "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=host;my-header1;x-amz-date, REDACTED_AWS_SECRET32890dbdd183b4468f858259531a5138ea",
    },
    {
        name: "get-header-value-multiline",
        request: {
            protocol: "https:",
            method: "GET",
            hostname: "example.amazonaws.com",
            query: {},
            headers: {
                host: "example.amazonaws.com",
                "my-header1": "value1,value2,value3",
                "x-amz-date": "20150830T123600Z",
            },
            path: "/",
        },
        authorization: "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=host;my-header1;x-amz-date, REDACTED_AWS_SECRETcc297cc0a3d70a5d00705980573d8ff790",
    },
    {
        name: "get-header-value-order",
        request: {
            protocol: "https:",
            method: "GET",
            hostname: "example.amazonaws.com",
            query: {},
            headers: {
                host: "example.amazonaws.com",
                "my-header1": "value4,value1,value3,value2",
                "x-amz-date": "20150830T123600Z",
            },
            path: "/",
        },
        authorization: "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=host;my-header1;x-amz-date, REDACTED_AWS_SECRETb1deb5e634ec47601a50643f830c755c01",
    },
    {
        name: "get-header-value-trim",
        request: {
            protocol: "https:",
            method: "GET",
            hostname: "example.amazonaws.com",
            query: {},
            headers: {
                host: "example.amazonaws.com",
                "my-header1": "value1",
                "my-header2": '"a   b   c"',
                "x-amz-date": "20150830T123600Z",
            },
            path: "/",
        },
        authorization: "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=host;my-header1;my-header2;x-amz-date, REDACTED_AWS_SECRET911fcaa05412b367055dee359757a9c736",
    },
    {
        name: "get-unreserved",
        request: {
            protocol: "https:",
            method: "GET",
            hostname: "example.amazonaws.com",
            query: {},
            headers: {
                host: "example.amazonaws.com",
                "x-amz-date": "20150830T123600Z",
            },
            path: "/-._~REDACTED_AWS_SECRETefghijklmnopqrstuvwxyz",
        },
        authorization: "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=host;x-amz-date, REDACTED_AWS_SECRET940f8a34d404d0cfa977f52a65bbf5f24f",
    },
    {
        name: "get-utf8",
        request: {
            protocol: "https:",
            method: "GET",
            hostname: "example.amazonaws.com",
            query: {},
            headers: {
                host: "example.amazonaws.com",
                "x-amz-date": "20150830T123600Z",
            },
            path: "/ሴ",
        },
        authorization: "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=host;x-amz-date, REDACTED_AWS_SECRETb787dc9c0e678f255a891fd03141be5d85",
    },
    {
        name: "get-vanilla",
        request: {
            protocol: "https:",
            method: "GET",
            hostname: "example.amazonaws.com",
            query: {},
            headers: {
                host: "example.amazonaws.com",
                "x-amz-date": "20150830T123600Z",
            },
            path: "/",
        },
        authorization: "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=host;x-amz-date, REDACTED_AWS_SECRET1e8372ff2a2260956d9b8aae1d763fbf31",
    },
    {
        name: "get-vanilla-empty-query-key",
        request: {
            protocol: "https:",
            method: "GET",
            hostname: "example.amazonaws.com",
            query: {
                Param1: "value1",
            },
            headers: {
                host: "example.amazonaws.com",
                "x-amz-date": "20150830T123600Z",
            },
            path: "/",
        },
        authorization: "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=host;x-amz-date, REDACTED_AWS_SECRETb97f1ea3c7a6bfe1b6e45aec72011b9aeb",
    },
    {
        name: "get-vanilla-query",
        request: {
            protocol: "https:",
            method: "GET",
            hostname: "example.amazonaws.com",
            query: {},
            headers: {
                host: "example.amazonaws.com",
                "x-amz-date": "20150830T123600Z",
            },
            path: "/",
        },
        authorization: "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=host;x-amz-date, REDACTED_AWS_SECRET1e8372ff2a2260956d9b8aae1d763fbf31",
    },
    {
        name: "get-vanilla-query-order-key-case",
        request: {
            protocol: "https:",
            method: "GET",
            hostname: "example.amazonaws.com",
            query: {
                Param2: "value2",
                Param1: "value1",
            },
            headers: {
                host: "example.amazonaws.com",
                "x-amz-date": "20150830T123600Z",
            },
            path: "/",
        },
        authorization: "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=host;x-amz-date, REDACTED_AWS_SECRET458b799221646efd99d3219ec94cdf2500",
    },
    {
        name: "get-vanilla-query-unreserved",
        request: {
            protocol: "https:",
            method: "GET",
            hostname: "example.amazonaws.com",
            query: {
                "-._~REDACTED_AWS_SECRETefghijklmnopqrstuvwxyz": "-._~REDACTED_AWS_SECRETefghijklmnopqrstuvwxyz",
            },
            headers: {
                host: "example.amazonaws.com",
                "x-amz-date": "20150830T123600Z",
            },
            path: "/",
        },
        authorization: "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=host;x-amz-date, REDACTED_AWS_SECRETcdf59bc7cc0884027167c21bb143a40197",
    },
    {
        name: "get-vanilla-utf8-query",
        request: {
            protocol: "https:",
            method: "GET",
            hostname: "example.amazonaws.com",
            query: {
                ሴ: "bar",
            },
            headers: {
                host: "example.amazonaws.com",
                "x-amz-date": "20150830T123600Z",
            },
            path: "/",
        },
        authorization: "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=host;x-amz-date, REDACTED_AWS_SECRET3c643bcf08f8b0a1d91e12c9027818dd04",
    },
    {
        name: "post-header-key-case",
        request: {
            protocol: "https:",
            method: "POST",
            hostname: "example.amazonaws.com",
            query: {},
            headers: {
                host: "example.amazonaws.com",
                "x-amz-date": "20150830T123600Z",
            },
            path: "/",
        },
        authorization: "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=host;x-amz-date, REDACTED_AWS_SECRET44621c30862966e37dddb68e92efbe5d6b",
    },
    {
        name: "post-header-key-sort",
        request: {
            protocol: "https:",
            method: "POST",
            hostname: "example.amazonaws.com",
            query: {},
            headers: {
                host: "example.amazonaws.com",
                "my-header1": "value1",
                "x-amz-date": "20150830T123600Z",
            },
            path: "/",
        },
        authorization: "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=host;my-header1;x-amz-date, REDACTED_AWS_SECRET6645f61f4dc9e1461ec8f8916fdf18852c",
    },
    {
        name: "post-header-value-case",
        request: {
            protocol: "https:",
            method: "POST",
            hostname: "example.amazonaws.com",
            query: {},
            headers: {
                host: "example.amazonaws.com",
                "my-header1": "VALUE1",
                "x-amz-date": "20150830T123600Z",
            },
            path: "/",
        },
        authorization: "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=host;my-header1;x-amz-date, REDACTED_AWS_SECRET7c5f22c7c4e8ae67b53629efa58b974b7d",
    },
    {
        name: "post-sts-header-after",
        request: {
            protocol: "https:",
            method: "POST",
            hostname: "example.amazonaws.com",
            query: {},
            headers: {
                host: "example.amazonaws.com",
                "x-amz-date": "20150830T123600Z",
            },
            path: "/",
        },
        authorization: "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=host;x-amz-date, REDACTED_AWS_SECRET44621c30862966e37dddb68e92efbe5d6b",
    },
    {
        name: "post-sts-header-before",
        request: {
            protocol: "https:",
            method: "POST",
            hostname: "example.amazonaws.com",
            query: {},
            headers: {
                host: "example.amazonaws.com",
                "x-amz-date": "20150830T123600Z",
                "x-amz-security-token": "REDACTED_AWS_SECRETREDACTED_AWS_SECRETREDACTED_AWS_SECRETREDACTED_AWS_SECRETREDACTED_AWS_SECRETREDACTED_AWS_SECRETREDACTED_AWS_SECRETREDACTED_AWS_SECRETQwj2ICCR/oLxBA==",
            },
            path: "/",
        },
        authorization: "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=host;x-amz-date;x-amz-security-token, REDACTED_AWS_SECRET0dd772bbebba041836c64533a82be05ead",
    },
    {
        name: "post-vanilla",
        request: {
            protocol: "https:",
            method: "POST",
            hostname: "example.amazonaws.com",
            query: {},
            headers: {
                host: "example.amazonaws.com",
                "x-amz-date": "20150830T123600Z",
            },
            path: "/",
        },
        authorization: "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=host;x-amz-date, REDACTED_AWS_SECRET44621c30862966e37dddb68e92efbe5d6b",
    },
    {
        name: "post-vanilla-empty-query-value",
        request: {
            protocol: "https:",
            method: "POST",
            hostname: "example.amazonaws.com",
            query: {
                Param1: "value1",
            },
            headers: {
                host: "example.amazonaws.com",
                "x-amz-date": "20150830T123600Z",
            },
            path: "/",
        },
        authorization: "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=host;x-amz-date, REDACTED_AWS_SECRETf1a96197d7deb8263271d420d138af7f11",
    },
    {
        name: "post-vanilla-query",
        request: {
            protocol: "https:",
            method: "POST",
            hostname: "example.amazonaws.com",
            query: {
                Param1: "value1",
            },
            headers: {
                host: "example.amazonaws.com",
                "x-amz-date": "20150830T123600Z",
            },
            path: "/",
        },
        authorization: "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=host;x-amz-date, REDACTED_AWS_SECRETf1a96197d7deb8263271d420d138af7f11",
    },
    {
        name: "post-vanilla-query-nonunreserved",
        request: {
            protocol: "https:",
            method: "POST",
            hostname: "example.amazonaws.com",
            query: {
                "@#$%^": "",
                "+": '/,?><`";:\\|][{}',
            },
            headers: {
                host: "example.amazonaws.com",
                "x-amz-date": "20150830T123600Z",
            },
            path: "/",
        },
        authorization: "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=host;x-amz-date, REDACTED_AWS_SECRETedc4c6df5ae71119d7cb98ed9b87393c1e",
    },
    {
        name: "post-vanilla-query-space",
        request: {
            protocol: "https:",
            method: "POST",
            hostname: "example.amazonaws.com",
            query: {
                p: "",
            },
            headers: {
                host: "example.amazonaws.com",
                "x-amz-date": "20150830T123600Z",
            },
            path: "/",
        },
        authorization: "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=host;x-amz-date, REDACTED_AWS_SECRETa3faa623b461c17f48f2fbdb9361b94a9b",
    },
    {
        name: "post-x-www-form-urlencoded",
        request: {
            protocol: "https:",
            method: "POST",
            hostname: "example.amazonaws.com",
            query: {},
            headers: {
                "content-type": "application/x-www-form-urlencoded",
                host: "example.amazonaws.com",
                "x-amz-date": "20150830T123600Z",
            },
            body: "Param1=value1",
            path: "/",
        },
        authorization: "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=content-type;host;x-amz-date, REDACTED_AWS_SECRETe5ac45fc38421b1da7b9d196a0fe09473a",
    },
    {
        name: "post-x-www-form-urlencoded-parameters",
        request: {
            protocol: "https:",
            method: "POST",
            hostname: "example.amazonaws.com",
            query: {},
            headers: {
                "content-type": "application/x-www-form-urlencoded; charset=utf8",
                host: "example.amazonaws.com",
                "x-amz-date": "20150830T123600Z",
            },
            body: "Param1=value1",
            path: "/",
        },
        authorization: "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=content-type;host;x-amz-date, REDACTED_AWS_SECRETce7fb2c7465f63e3092b3b0d39fa77a6fe",
    },
];
